pulser/simulation/simulation.py

```diff
@@ -640,9 +677,53 @@ def make_xy_term() -> qutip.Qobj:
                 )
             return xy
 
-        make_interaction_term = (
-            make_xy_term if self._interaction == "XY" else make_vdw_term
-        )
+        def make_masked_xy_term() -> qutip.Qobj:
```

Comment:
There's an awful lot of code replication from this function. Why not just modifiy the `make_xy_term` function to accept a `masked=True/False` keyword argument instead?

Reply:
You're right, I'll do that